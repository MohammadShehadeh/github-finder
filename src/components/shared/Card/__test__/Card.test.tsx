import renderer from 'react-test-renderer';

import { Card } from '..';

const mockCardProps = {
  url: 'url',
  language: 'js',
  forksCount: 10,
  starsCount: 10,
};

describe('Card', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<Card {...mockCardProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render to match snapshot with missing data', () => {
    const tree = renderer
      .create(<Card {...mockCardProps} language="" forksCount={0} starsCount={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
