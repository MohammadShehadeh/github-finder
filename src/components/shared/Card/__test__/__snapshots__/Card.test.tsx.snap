// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render to match snapshot 1`] = `
<a
  className="card"
  href="url"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Github Profile Image"
    data-nimg="1"
    decoding="async"
    height={100}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fplaceholder.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplaceholder.png&w=128&q=75 1x, /_next/image?url=%2Fplaceholder.png&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <h3
    className="name"
  >
    Github Profile Image
  </h3>
  <span
    className="eyebrow language contained secondary"
  >
    <span
      className="label"
    >
      js
    </span>
  </span>
  <span
    className="eyebrow forks contained tertiary"
  >
    <span
      className="label"
    >
      Fork: 10
    </span>
  </span>
  <span
    className="eyebrow stars contained tertiary"
  >
    <span
      className="label"
    >
      Star: 10
    </span>
  </span>
</a>
`;

exports[`Card should render to match snapshot with missing data 1`] = `
<a
  className="card"
  href="url"
  rel="noopener noreferrer"
  target="_blank"
>
  <img
    alt="Github Profile Image"
    data-nimg="1"
    decoding="async"
    height={100}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fplaceholder.png&w=256&q=75"
    srcSet="/_next/image?url=%2Fplaceholder.png&w=128&q=75 1x, /_next/image?url=%2Fplaceholder.png&w=256&q=75 2x"
    style={
      {
        "color": "transparent",
      }
    }
    width={100}
  />
  <h3
    className="name"
  >
    Github Profile Image
  </h3>
</a>
`;
